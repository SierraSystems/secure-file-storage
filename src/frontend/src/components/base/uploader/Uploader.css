.custom-width {
  width: 60%;
  margin: auto;
}

@media only screen and (max-device-width: 480px) {
  .custom-width {
    width: 80%;
    margin: auto;
    margin-bottom: 50px;
  }
}
