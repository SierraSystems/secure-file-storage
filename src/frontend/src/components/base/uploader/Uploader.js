import React, { useState } from "react";
// Import React FilePond
import { FilePond, registerPlugin } from "react-filepond";

// Import FilePond styles
import "filepond/dist/filepond.min.css";

// Import the Image EXIF Orientation and Image Preview plugins
import FilePondPluginImageExifOrientation from "filepond-plugin-image-exif-orientation";
import FilePondPluginImagePreview from "filepond-plugin-image-preview";
import "filepond-plugin-image-preview/dist/filepond-plugin-image-preview.css";

// Register the plugins
registerPlugin(FilePondPluginImageExifOrientation, FilePondPluginImagePreview);

export default function Uploader() {
  const [files, setFiles] = useState([]);

  return (
    <div className="App">
      <FilePond
        files={files}
        allowMultiple
        onupdatefiles={setFiles}
        labelIdle='Drag and Drop your files or <span class="filepond--label-action">Browse</span>'
      />
    </div>
  );
}
