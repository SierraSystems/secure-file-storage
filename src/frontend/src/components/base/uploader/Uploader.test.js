import React from "react";
import Uploader from "./Uploader";
import testBasicSnapshot from "../../../TestHelper";

describe("Uploader", () => {
  test("Component renders as expected", () => {
    testBasicSnapshot(<Uploader />);
  });
});
