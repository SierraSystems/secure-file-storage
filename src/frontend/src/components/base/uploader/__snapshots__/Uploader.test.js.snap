// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader Component renders as expected 1`] = `
<div
  className="spacing custom-width"
>
  <div
    className="filepond--wrapper"
  >
    <input
      multiple={true}
      type="file"
    />
  </div>
</div>
`;
